### PR TITLE
[15.0][MIG] purchase_tier_validation

### PR DESCRIPTION
**Pull request by JasminSForgeFlow**
Standard Migration
@ForgeFlow

### CODE REVIEW

**Comment by etobella** [comment]
Tests are failing. Can you rebase and try again?

**Comment by JasminSForgeFlow** [comment]
> Tests are failing. Can you rebase and try again?

Done, Thanks

File: purchase_tier_validation/tests/common.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2018 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+def setup_test_model(env, model_clses):
```
Comment (simahawk) [inline purchase_tier_validation/tests/common.py]: this approach is deprecated. Can you please use https://github.com/OCA/odoo-test-helper ?

File: purchase_tier_validation/tests/test_tier_validation.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+from .common import setup_test_model, teardown_test_model
+from .tier_validation_tester import TierValidationTester
+
+
+@tagged("post_install", "-at_install")
+class TestPurchaseTierValidation(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestPurchaseTierValidation, cls).setUpClass()
```
Comment (simahawk) [inline purchase_tier_validation/tests/test_tier_validation.py:14]: can you set `tracking_disable` on the env?

File: purchase_tier_validation/tests/test_tier_validation.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                "definition_domain": "[('test_field', '>', 1.0)]",
+            }
+        )
+
+        cls.test_record = cls.test_model.create({"test_field": 2.5})
+
+    @classmethod
+    def tearDownClass(cls):
+        teardown_test_model(cls.env, [TierValidationTester])
+        return super(TestPurchaseTierValidation, cls).tearDownClass()
```
Comment (simahawk) [inline purchase_tier_validation/tests/test_tier_validation.py:63]: no need to return anything here

**Review by simahawk — COMMENTED** [review]
@JasminSForgeFlow only technical remarks from me.
If you want to tackle them in a second PR, I won't block a PR that is here since so long for these small reasons (if you promise you'll take care :stuck_out_tongue_winking_eye: )

**Review by Camille0907 — APPROVED** [review]
LGTM, thanks

File: purchase_tier_validation/tests/common.py (hunk 1 above)
Comment (JasminSForgeFlow) [reply purchase_tier_validation/tests/common.py]: Updated, Thanks

File: purchase_tier_validation/tests/test_tier_validation.py (hunk 2 above)
Comment (JasminSForgeFlow) [reply purchase_tier_validation/tests/test_tier_validation.py:14]: Updated, Thanks

File: purchase_tier_validation/tests/test_tier_validation.py (hunk 3 above)
Comment (JasminSForgeFlow) [reply purchase_tier_validation/tests/test_tier_validation.py:63]: Pre-commit is failing if not add return

File: purchase_tier_validation/tests/test_tier_validation.py (hunk 2 above)
Comment (simahawk) [reply purchase_tier_validation/tests/test_tier_validation.py:14]: it won't work like that. You must override it in setUpClass like
```
        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```

File: purchase_tier_validation/tests/test_tier_validation.py (hunk 3 above)
Comment (simahawk) [reply purchase_tier_validation/tests/test_tier_validation.py:63]: then pre-commit is wrong and you can work around it w/ `# pylint: disable=$err_code`

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by JasminSForgeFlow** [comment]
@simahawk Changes have been updated.

Thanks

File: purchase_tier_validation/tests/test_tier_validation.py (hunk 3 above)
Comment (simahawk) [reply purchase_tier_validation/tests/test_tier_validation.py:63]: this has not changed. I won't block the PR but please rewrite this w/ disable comment in a new PR to not encourage this approach.

**Review by simahawk — APPROVED** [review]

**Pull request merged**